components/bpmn/org.wso2.carbon.bpmn.analytics.publisher/src/main/java/org/wso2/carbon/bpmn/analytics/publisher/AnalyticsPublisher.java

```diff
@@ -0,0 +1,273 @@
+	 */
+	private void doPollingForInstances() {
+		log.debug("Start polling for process instances...");
+		try {
+			Thread.sleep(AnalyticsPublisherConstants.DELAY);
+			while (true) {
+				BPMNProcessInstance[] bpmnProcessInstances =
+						analyticsPublishServiceUtils.getCompletedProcessInstances();
+				if (bpmnProcessInstances != null && bpmnProcessInstances.length > 0) {
+					for (BPMNProcessInstance instance : bpmnProcessInstances) {
+						publishBPMNProcessInstanceEvent(instance);
+					}
+				}
+				BPMNTaskInstance[] bpmnTaskInstances =
+						analyticsPublishServiceUtils.getCompletedTaskInstances();
+				if (bpmnTaskInstances != null && bpmnTaskInstances.length > 0) {
+					for (BPMNTaskInstance instance : bpmnTaskInstances) {
+						publishBPMNTaskInstanceEvent(instance);
+					}
+				}
+				Thread.sleep(AnalyticsPublisherConstants.REPEATEDLY_DELAY);
+			}
+		} catch (AgentException e) {
+			String errMsg = "Agent exception in polling thread for BPMN process instances.";
+			log.error(errMsg, e);
+		} catch (InterruptedException e) {
+			String errMsg = "I/O exception in polling thread for BPMN process instances.";
+			log.error(errMsg, e);
+		}
+	}
+
+	/**
+	 * Publish process instance as events to the data receiver
+	 *
+	 * @param bpmnProcessInstance BPMN process instance to retrieve the data for payload param of data publisher's publish method
+	 * @throws AgentException
+	 */
+	private void publishBPMNProcessInstanceEvent(BPMNProcessInstance bpmnProcessInstance)
+			throws AgentException {
+		Object[] payload = new Object[] { bpmnProcessInstance.getProcessDefinitionId(),
+		                                  bpmnProcessInstance.getInstanceId(),
+		                                  bpmnProcessInstance.getStartTime().toString(),
+		                                  bpmnProcessInstance.getEndTime().toString(),
+		                                  bpmnProcessInstance.getDuration() };
+		log.debug("Start to Publish BPMN process instance event...");
+		dataPublisher.publish(processInstanceStreamId, getMeta(), null, payload);
+		log.debug("Published BPMN process instance event...");
+	}
+
+	/**
+	 * Publish task instance as events to the data receiver
+	 *
+	 * @param bpmnTaskInstance BPMN task instance to retrieve the data for payload param of data publisher's publish method
+	 * @throws AgentException
+	 */
+	private void publishBPMNTaskInstanceEvent(BPMNTaskInstance bpmnTaskInstance)
+			throws AgentException {
+		Object[] payload = new Object[] { bpmnTaskInstance.getTaskDefinitionKey(),
+		                                  bpmnTaskInstance.getTaskInstanceId(),
+		                                  bpmnTaskInstance.getStartTime().toString(),
+		                                  bpmnTaskInstance.getEndTime().toString(),
+		                                  bpmnTaskInstance.getDurationInMills(),
+		                                  bpmnTaskInstance.getAssignee() };
+		log.debug("Start to Publish BPMN task instance event...");
```

Comment:
can we add some usefull info for this debug logs here, such as instance id. If you are adding them, you will have to add isDebugEnabled if block also.